app/src/main/java/com/github/swent/echo/data/repository/datasources/LocalDataSource.kt

```diff
@@ -1,5 +1,53 @@
 package com.github.swent.echo.data.repository.datasources
 
+import com.github.swent.echo.data.model.Association
+import com.github.swent.echo.data.model.Event
+import com.github.swent.echo.data.model.Tag
+import com.github.swent.echo.data.model.UserProfile
+
+/**
+ * The local data source.
+ *
+ * Note that all the getters return null when the data is not found locally.
+ */
 interface LocalDataSource {
-    fun getData(): String
+
+    /** Initialize the local data source. */
+    suspend fun initialize()
```

Comment:
I'm wondering if there is any interest in having this here. Anyways initializing the DB should be a one-time action that's never repeated. Does it make sense for the consumer of the LocalDataSource (i.e. the Repository) to care about whether Room is initialized? Would it maybe make more sense to not expose this and just have room initialize itself either on the first startup of the app, a hook during login/registration or simply when one of it's other methods is called for the first time ?

Reply:
You're right! I added this as I thought we might need to do some initialization logic, but after starting implementing the interface and reading through the docs of room, I didn't find anything that should be initialized... Therefore, I removed it.

Reply:
okay great :)